database/upgrades/16-refactor-postgres.sql

```diff
@@ -48,7 +51,7 @@ INSERT INTO lost_portals SELECT mxid, chat_id, receiver FROM portal WHERE mxid<>
 UPDATE portal SET mxid=NULL WHERE mxid='';
 ALTER TABLE portal ADD CONSTRAINT portal_mxid_unique UNIQUE(mxid);
 -- Delete any portals that aren't associated with logged-in users.
-DELETE FROM portal WHERE receiver<>'' AND receiver NOT IN (SELECT username FROM "user" WHERE uuid IS NOT NULL);
+DELETE FROM portal WHERE receiver<>'' AND receiver NOT IN (SELECT username FROM "user" WHERE username IS NOT NULL AND uuid IS NOT NULL);
```

Comment:
This should address #444.  Without it, the `NOT IN` will match no rows if any null username exists, which would prevent any rows from being deleted.  That behaviour of `NOT IN` is explained by https://modern-sql.com/concept/three-valued-logic#not-in-null.

Reply:
Do you mean #425?

Reply:
Ah, yes

Reply:
Hmm, looks like this can hang...  I'll go back to the drawing board for this

Reply:
This was correct after all, but was triggering some slow CASCADE actions on our big database.